### DISCUSSION TITLE
打程式要使用什麼輸入法

### DISCUSSION THREAD

**Discussion by enoch-lightning** [1101-GS4719-C/Python-Community-GS4719-C#174] [General]
因為所有程式都使用英文撰寫(好像也可以叫作「編程」？)，因此打程式時，必須使用電腦本身的英文輸入法，並且為半形模式。
# 怎麼確認電腦目前是什麼輸入模式？
## 中文輸入法
<img width="960" alt="中文輸入法" src="https://user-images.githubusercontent.com/86505454/144047878-c8ffeae9-3eb3-44aa-9ec5-d4705595e0e2.png">

## 英文輸入法
<img width="960" alt="英文輸入法" src="https://user-images.githubusercontent.com/86505454/144048324-eab68f62-83d3-41ad-a7e2-ef9fc6cd6106.png">

## 如何在中文輸入法，與英文輸入法間切換
1.以滑鼠左鍵點擊顯示輸入法的地方一下
顯示輸入法的地方，就是我在上面標題「中文輸入法」與「英文輸入法」，用紅色方框框起來的地方。
2.按電腦鍵盤上的shift(在開啟網頁的時候，有時候不能在中文輸入法，與英文輸入法間切換，會固定在中文輸入法。我不知道為什麼會這樣。)

### 如何知道英文輸入法，是處在「全形」模式，還是「半形」模式下
1.在電腦是英文輸入法的情況下，將滑鼠移到顯示輸入法的地方，也就是紅色方框框起來的地方。
<img width="960" alt="英文輸入法 確認為全形還是半形第一步" src="https://user-images.githubusercontent.com/86505454/144077297-92a4eacc-93de-4a1c-b402-db395077709b.png">
2.在紅色方框框起來的地方，也就是「英」，點選滑鼠右鍵。
<img width="960" alt="英文輸入法 確認為全形還是半形第二步" src="https://user-images.githubusercontent.com/86505454/144077870-e39e9840-f70b-4874-9a90-359be73be2ad.png">
3.看「全形/半形切換 (目前是全形還是半形) (W)」，就可以知道目前是全形還是半形
<img width="960" alt="英文輸入法 確認為全形還是半形第三步" src="https://user-images.githubusercontent.com/86505454/144079000-7c4e1283-2c44-4393-94ea-3502ffacf8da.png">
4.將「滑鼠」移到「全形/半形切換 (目前是全形還是半形) (W)」，就可以看到切換全形與半形的選單
[嘗試解說影片](https://user-images.githubusercontent.com/86505454/144081140-22c16b53-d82e-4f19-9935-68d250baf02b.mp4)
### 英文輸入法要如何切換全形與半形？？
按shift+空白=空格=space
<img width="858" alt="這是「空白鍵」" src="https://user-images.githubusercontent.com/86505454/149605865-318367db-b818-4f0b-9a59-ea3349e9ea4e.png">

**Comment by xCk27x** [comment]
shift+space 能切換全半形真的超方便
打字常常會需要轉換 終於不用在輸入法選單那邊耗時間了

**Reply by enoch-lightning** [reply]
我後來發現是在可以打字的地方，才能用點選的方式在中文輸入法與英文輸入法下，切換全形與半形。

**Reply by enoch-lightning** [reply]
要切換中文與英文輸入法，可以直接用滑鼠/觸控板左鍵點選輸入法標示，也可以按shift。要切換全形與半形，可以在輸入法標示上點滑鼠/觸控板右鍵，在「全形/半形切換」以滑鼠/觸控板左鍵點選切換，也可以按「shift+空白」切換。

**Comment by 109802503gloria** [comment]
沒想到會有人專門做一個討論區教大家使用字型切換，還有教學影片可以看，同學太用心了

**Comment by 107602505** [comment]
在看這篇文章之前根本不知道原來英文輸入法還有分成全形跟半形算是長知識了哈哈，特別查了下原來「全形」是指文字字身長寬比為一比一的正方形，而「半形」為寬度為全形一半的文字。下面是全形和半形的差別
全形：ＡＢＣＤＥＦＧ
半形：ABCDEFG

**Comment by enoch-lightning** [comment]
我也去查了資料，了解到「半形」在螢幕上佔據的空間，是一個字元(char)的位置，而「全形」在螢幕上佔據的空間，是兩個字元的位置。「字元」是一種變數的型別，也就是一種資料型態。在Python裡沒有「字元」這種資料型態，但在C++裡面有。在C++中，一個字元佔用一個位元組(byte)，也就是八個位元(bits)。1byte=8bits

**Reply by enoch-lightning** [reply]
中文字一定是使用「全形」，原因如下

1. 儲存一個字元(char)的記憶體空間為1byte=8bits。每一個bit都只能存0或1，8bits只能儲存2的8次方=256個數字。但中文字很複雜，沒辦法用256個數字表示一個字，也就是不能用一個字元儲存一個中文字。必須要使用兩個字元，也就是2的8次方*2的8次方=2的16次方個數字去表示一個中文字。使用兩個字元儲存就是全形。
2. 問題：可以不要使用兩個字元，只使用1.xxx個字元儲存中文字嗎？
答案：不行。因為電腦在使用記憶體時，使用的最小單位是1byte=8bits。因此沒有任何東西的儲存空間可以分配到1.xxx byte，而1byte就是一個字元，也就是不可能會有(1 byte/字元)(1.xxx 字元)=1.xxx bytes，不可能有1.xxx 個字元。
- 位元：英文單數(一個)為bit，複數(兩個、三個、四個或更多個)為bits。8bits就是8個bit。
- *就是Python中加減乘除的乘(數學中2x3=6的x)。所有的程式語言中，*好像都是乘，就是相乘。

**Reply by enoch-lightning** [reply]
ASCII碼就是半形。

**Comment by enoch-lightning** [comment]
如果覺得我講錯，非常歡迎提出指正。
🌞

**Comment by ghost** [comment]
全形半形...有時候密碼裡有標點符號的時候，像是這兩個驚嘆號式不一樣的 (!) (！)
然後就會密碼錯誤，易發生於平板上設密碼的時候，沒注意切到的驚嘆號式全形還是半形就會出問題

**Comment by cody910120** [comment]
提醒一下玩賣快的時候記得切英文輸入法喔，不然案到shift你會卡住要再切一次輸入法，有夠麻煩

**Comment by YuHengChen54** [comment]
我發現一個酷酷的東西，我把左邊shift切換輸入法關掉，可是用左邊shift+space還是可以切換全形半形

**Comment by 109302013evantsao** [comment]
提醒一下各位，快速切換輸入法可以使用shift+alt進行使用，方便速速在中英日文中間做轉換

**Reply by enoch-lightning** [reply]
我按shift+alt(先按住shift不要放，再按alt)輸入法不會切換，不知道為什麼。
